SYCL/Plugin/level-zero-event-leak.cpp

```diff
@@ -0,0 +1,42 @@
+// REQUIRES: level_zero, level_zero_dev_kit
+// RUN: %clangxx -fsycl -fsycl-unnamed-lambda -fsycl-targets=%sycl_triple %level_zero_options %s -o %t.out
+// RUN: env SYCL_DEVICE_FILTER=level_zero ZE_DEBUG=4 %GPU_RUN_PLACEHOLDER %t.out
+//
+// CHECK-NOT: ---> LEAK
+
+// The test is to check that there are no leaks reported with the embedded
+// ZE_DEBUG=4 testing capability. Example of a leak reported is this:
+//
+// clang-format off
+// ZE_DEBUG=4: check balance of create/destroy calls
+// ----------------------------------------------------------
+//               zeContextCreate = 1     \--->              zeContextDestroy = 1
+//          zeCommandQueueCreate = 1     \--->         zeCommandQueueDestroy = 0     ---> LEAK = 1
+//                zeModuleCreate = 1     \--->               zeModuleDestroy = 0     ---> LEAK = 1
+//                zeKernelCreate = 1     \--->               zeKernelDestroy = 0     ---> LEAK = 1
+//             zeEventPoolCreate = 1     \--->            zeEventPoolDestroy = 1
+//  zeCommandListCreateImmediate = 1     |
+//           zeCommandListCreate = 2     \--->          zeCommandListDestroy = 2     ---> LEAK = 1
+//                 zeEventCreate = 129   \--->                zeEventDestroy = 1     ---> LEAK = 128
+//                 zeFenceCreate = 2     \--->                zeFenceDestroy = 0     ---> LEAK = 2
+//                 zeImageCreate = 0     \--->                zeImageDestroy = 0
+//               zeSamplerCreate = 0     \--->              zeSamplerDestroy = 0
+//              zeMemAllocDevice = 0     |
+//                zeMemAllocHost = 0     |
+//              zeMemAllocShared = 0     \--->                     zeMemFree = 0
+//
+// clang-format on
+//
+// NOTE: The 129 value below is to be larger thatn the "128" heuristic in
```

Comment:
If the 128 heuristic in addSharedEvernt is changed to be larger (which it probably should be) then will this test
need to be updated?

Should the value be chosen to be MUCH larger than the current heuristic?

Reply:
OK, put 1000.